RecoVertex/PrimaryVertexProducer/plugins/PrimaryVertexProducer.cc

```diff
@@ -115,6 +115,13 @@ PrimaryVertexProducer::PrimaryVertexProducer(const edm::ParameterSet& conf) : th
     algorithms.push_back(algorithm);
     produces<reco::VertexCollection>(algorithm.label);
   }
+
+  //check if this is a recovery iteration
+  fRecoveryIteration = conf.getParameter<bool>("isRecoveryIteration");
+  if (fRecoveryIteration) {
+    assert(algorithms.size() == 1);
```

Comment:
To report configuration mistakes an (explanatory) exception would be easier to understand than assertion failure.